main.js

```diff
@@ -59,162 +47,169 @@ const dice = {
     }
   },
 
-  /**
-   * Remove 'select' class from all dice
-   */
   removeAllSelect: function () {
     for (const die of this.positions) {
       dicePositions[die].classList.remove("select");
     }
   },
-};
 
-// --- SELECT --- \\
-
-// Event listers added on all dice for toggling the 'select' class
-for (const die of dice.positions) {
-  dicePositions[die].addEventListener("click", () => {
-    if (dice.shouldRoll[die] && dice.positionValues[0] !== 0) {
-      dicePositions[die].classList.toggle("select");
-
-      points.selectedTotal = 0;
-      points.calculate();
-      uncommitedPoints.innerHTML = points.turnTotal + points.selectedTotal;
+  /**
+   * Update shouldRoll from tempShouldRoll during scoring
+   */
+  updateShouldRoll: function () {
+    for (const die of dice.positions) {
+      if (
+        dice.shouldRoll[die] === true &&
+        scoring.tempShouldRoll[die] === false
+      ) {
+        dice.shouldRoll[die] = false;
+      }
     }
-  });
-}
-
-// --- HOLD --- \\
-
-// Hold Dice - Switch player
-document.getElementById("hold-button").addEventListener("click", () => {
-  if (activePlayer1.classList[1] === "active") {
-    scorePlayer1.innerHTML = points.turnTotal + points.selectedTotal;
-  } else {
-    scorePlayer2.innerHTML = points.turnTotal + points.selectedTotal;
-  }
-
-  activePlayer1.classList.toggle("active");
-  activePlayer2.classList.toggle("active");
-
-  points.turnTotal = 0;
-  points.selectedTotal = 0;
-  uncommitedPoints.innerHTML = 0;
-
-  dice.removeAllSelect();
-  dice.positionValues = [0, 0, 0, 0, 0, 0];
-  dice.shouldRoll = [true, true, true, true, true, true];
-  dice.display();
-});
+  },
+};
 
-// --- SCORE --- \\
+// --- POINTS --- \\
 
 points = {
   player1: 0,
   player2: 0,
   turnTotal: 0,
-  selectedTotal: 0,
 
-  calculate: function () {
-    diceSelected = this.diceSelected();
+  updateDisplay: function (selectedTotal) {
+    if (scoring.selectedTotal === 0) {
+      this.turnTotal = 0;
+      selectedTotal = 0;
+    }
 
-    freqObj = this.freq(diceSelected);
-    console.log(freqObj);
+    if (activePlayer1.classList.contains("active")) {
+      points.player1 += this.turnTotal + selectedTotal;
+      scorePlayer1.innerHTML = points.player1;
+    } else {
+      if (scoring.selectedTotal > 0) {
+        points.player2 += this.turnTotal + selectedTotal;
+        scorePlayer2.innerHTML = points.player2;
+      }
+    }
+  },
+};
+
+// --- SCORING --- \\
 
-    this.scoring(freqObj);
+scoring = {
+  selectedTotal: 0,
+  tempShouldRoll: [true, true, true, true, true, true],
+
+  calculate: function () {
+    diceSelectedArr = this.diceSelected();
+    freqObj = this.freq(diceSelectedArr);
+    this.readDiceCombos(freqObj);
   },
 
   diceSelected: function () {
-    diceSelected = [];
+    diceSelectedArr = [];
     for (const die of dice.positions) {
       if (
-        dicePositions[die].classList[1] === "select" &&
+        dicePositions[die].classList.contains("select") &&
```

Comment:
Shouldn't it be `includes`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
I tried includes() and it didn't work. Googlized it and found that includes() works for arrays, but for nodes contains() is needed.

Reply:
Sweet